src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CallVariantsFromAlignedContigsSpark.java

```diff
@@ -0,0 +1,382 @@
+
+                final BreakpointAlignment breakpointAlignment = new BreakpointAlignment(current.contigId, previous, current, insertedSequence, homology, insertionAlignmentRegions);
+
+                results.add(breakpointAlignment);
+            }
+        }
+        return results;
+    }
+
+    private static String getInsertedSequence(final AlignmentRegion current, final AlignmentRegion previous, final byte[] sequenceCopy) {
+        String insertedSequence = "";
+        if (previous.endInAssembledContig < current.startInAssembledContig - 1) {
+
+            final int insertionStart;
+            final int insertionEnd;
+
+            insertionStart = previous.endInAssembledContig + 1;
+            insertionEnd = current.startInAssembledContig - 1;
+
+            final byte[] insertedSequenceBytes = Arrays.copyOfRange(sequenceCopy, insertionStart - 1, insertionEnd);
+            if (previous.referenceInterval.getStart() > current.referenceInterval.getStart()) {
+                SequenceUtil.reverseComplement(insertedSequenceBytes, 0, insertedSequenceBytes.length);
+            }
+            insertedSequence = new String(insertedSequenceBytes);
+        }
+        return insertedSequence;
+    }
+
+    private static String getHomology(final AlignmentRegion current, final AlignmentRegion previous, final byte[] sequenceCopy) {
+        String homology = "";
+        if (previous.endInAssembledContig >= current.startInAssembledContig) {
+            final byte[] homologyBytes = Arrays.copyOfRange(sequenceCopy, current.startInAssembledContig - 1, previous.endInAssembledContig);
+            if (previous.referenceInterval.getStart() > current.referenceInterval.getStart()) {
+                SequenceUtil.reverseComplement(homologyBytes, 0, homologyBytes.length);
+            }
+            homology = new String(homologyBytes);
+        }
+        return homology;
+    }
+
+    private static boolean currentAlignmentRegionIsTooSmall(final AlignmentRegion current, final AlignmentRegion next, final Integer minAlignLength) {
+        return current.referenceInterval.size() - current.overlapOnContig(next) < minAlignLength;
+    }
+
+    protected static boolean treatNextAlignmentRegionInPairAsInsertion(AlignmentRegion current, AlignmentRegion next, final Integer minAlignLength) {
+        return treatAlignmentRegionAsInsertion(next) ||
+                (next.referenceInterval.size() - current.overlapOnContig(next) < minAlignLength) ||
+                current.referenceInterval.contains(next.referenceInterval) ||
+                next.referenceInterval.contains(current.referenceInterval);
+    }
+
+    private static boolean treatAlignmentRegionAsInsertion(final AlignmentRegion next) {
+        return next.mqual < 60;
+    }
+
+    public static void writeVariants(final String fastaReference, final Logger logger, final JavaRDD<VariantContext> variantContexts, final PipelineOptions pipelineOptions, final String outputPath) {
```

Comment:
It seems that this method could also be called within the public method (which itself could become package-access) `callVariantsFromAlignmentRegions` (and of course its name could be changed then to reflect the fact that it is also writing output).
It you decide this is worth it, then `writeVariants` could become private as well.